posthog/models/person/person.py

```diff
@@ -65,9 +67,11 @@ def split_person(self, main_distinct_id: Optional[str], max_splits: Optional[int
             if not distinct_id == main_distinct_id:
                 with transaction.atomic():
                     pdi = PersonDistinctId.objects.select_for_update().get(person=self, distinct_id=distinct_id)
-                    person = Person.objects.create(team_id=self.team_id)
+                    person, _ = Person.objects.get_or_create(
+                        uuid=uuidFromDistinctId(self.team_id, distinct_id), team_id=self.team_id
```

Comment:
I need to set `persion.version` still, and write tests because this is getting hairy.